File: Python/485.max-consecutive-ones.py
#
# @lc app=leetcode id=485 lang=python
#
# [485] Max Consecutive Ones
#
# https://leetcode.com/problems/max-consecutive-ones/description/
#
# algorithms
# Easy (55.72%)
# Likes:    432
# Dislikes: 336
# Total Accepted:    159.4K
# Total Submissions: 285.7K
# Testcase Example:  '[1,0,1,1,0,1]'
#
# Given a binary array, find the maximum number of consecutive 1s in this
# array.
# 
# Example 1:
# 
# Input: [1,1,0,1,1,1]
# Output: 3
# Explanation: The first two digits or the last three digits are consecutive
# 1s.
# ⁠   The maximum number of consecutive 1s is 3.
# 
# 
# 
# Note:
# 
# The input array will only contain 0 and 1.
# The length of input array is a positive integer and will not exceed 10,000
# 
# 
#

# @lc code=start
class Solution(object):
    def findMaxConsecutiveOnes(self, nums):
        """
        :type nums: List[int]
        :rtype: int
        """
        ans = 0
        count = 0
        for num in nums:
            if num == 1:
                count += 1
            else:
                ans = max(ans, count)
                count = 0
        return max(ans, count)
        
# @lc code=end

File: C++/485.max-consecutive-ones.cpp
/*
 * @lc app=leetcode id=485 lang=cpp
 *
 * [485] Max Consecutive Ones
 *
 * https://leetcode.com/problems/max-consecutive-ones/description/
 *
 * algorithms
 * Easy (56.07%)
 * Likes:    493
 * Dislikes: 346
 * Total Accepted:    171.7K
 * Total Submissions: 306.2K
 * Testcase Example:  '[1,0,1,1,0,1]'
 *
 * Given a binary array, find the maximum number of consecutive 1s in this
 * array.
 * 
 * Example 1:
 * 
 * Input: [1,1,0,1,1,1]
 * Output: 3
 * Explanation: The first two digits or the last three digits are consecutive
 * 1s.
 * ⁠   The maximum number of consecutive 1s is 3.
 * 
 * 
 * 
 * Note:
 * 
 * The input array will only contain 0 and 1.
 * The length of input array is a positive integer and will not exceed 10,000
 * 
 * 
 */
#include<vector>
using namespace std;

// @lc code=start
class Solution {
public:
    int findMaxConsecutiveOnes(vector<int>& nums) {
        // int ans = 0;
        // for (int i=0, last=-1; i<nums.size(); i++) {
        //     if (nums[i]==0)
        //         last = i;
        //     else
        //         ans = max(ans, i-last);
        // }   
        // return ans;
        
        int ans = 0, cnt = 0;
        for (int num: nums) {
            if (num==1) {
                cnt ++;
                ans = max(ans, cnt);
            }
            else 
                cnt = 0;
        }
        return ans;
    }
};
// @lc code=end

